### PR TITLE
A few changes to support msvc 2017

### PR DESCRIPTION
**Pull request by LamarMoore**
Fixes warnings coming from the MSVC 2017 build.

**To test:**

Build locally using msvc 2017 to ensure everything builds cleanly and they are no build or warnings. Also ensure the 2015 build still works,

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/MDAlgorithms/src/CalculateCoverageDGS.cpp [hunk 1]
```diff
@@ -425,7 +425,9 @@ void CalculateCoverageDGS::exec() {
       std::vector<coord_t> pos(4);
       std::transform(curIntSec.getBareArray(), curIntSec.getBareArray() + 4,
                      prevIntSec.getBareArray(), pos.begin(),
-                     VectorHelper::SimpleAverage<double>());
```
Comment (martyngigg) [inline Framework/MDAlgorithms/src/CalculateCoverageDGS.cpp]: Does `SimpleAverage` produce warnings now?

Comment (LamarMoore) [reply Framework/MDAlgorithms/src/CalculateCoverageDGS.cpp]: Yes because the `double` return value is being converted to a `float` (`coord_t`) in `pos`.

**Comment by martyngigg** [comment]
This will now be done in the maintenance period after 3.13.

File: Framework/MDAlgorithms/src/CalculateCoverageDGS.cpp (hunk 1 above)
Comment (martyngigg) [reply Framework/MDAlgorithms/src/CalculateCoverageDGS.cpp]: Should we remove the `SimpleAverage` functor now?

File: buildconfig/CMake/FindBoost.cmake [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
```
Comment (martyngigg) [inline buildconfig/CMake/FindBoost.cmake]: If we upgrade cmake as discussed can we remove this file?

File: buildconfig/CMake/FindBoostPython.cmake [hunk 3]
```diff
@@ -5,7 +5,11 @@
 #  - windows: boost_python (python2), ????? (python3)
 #  - others?
 if ( MSVC )
-  find_package ( Boost COMPONENTS python REQUIRED )
+  if ( ${Boost_VERSION} LESS 106700)
```
Comment (martyngigg) [inline buildconfig/CMake/FindBoostPython.cmake]: Maybe we don't need the `if` check? We could assume that on windows we always know what version of boost we're running.

Comment (LamarMoore) [reply buildconfig/CMake/FindBoostPython.cmake]: Good point

File: buildconfig/CMake/FindBoost.cmake (hunk 2 above)
Comment (LamarMoore) [reply buildconfig/CMake/FindBoost.cmake]: Won't it still be needed for other platforms? Or is this strictly used for windows?

Comment (martyngigg) [reply buildconfig/CMake/FindBoost.cmake]: Good question. I misunderstood and thought we added it here. I'm not sure why we had it in the first place could you ask on Slack if anyone knows?

**Review by martyngigg — APPROVED** [review]

**Pull request merged**